packages/core/saved-objects/core-saved-objects-api-server-internal/src/lib/apis/get.ts

```diff
@@ -84,18 +72,8 @@ export const performGet = async <T>(
     migrationVersionCompatibility,
   });
 
-  let migrated: SavedObject<T>;
-  try {
-    migrated = migrationHelper.migrateStorageDocument(document) as SavedObject<T>;
-  } catch (error) {
-    throw SavedObjectsErrorHelpers.decorateGeneralError(
-      error,
-      'Failed to migrate document to the latest version.'
-    );
-  }
-
-  return encryptionHelper.optionallyDecryptAndRedactSingleResult(
-    migrated,
-    authorizationResult?.typeMap
-  );
+  return await migrationHelper.migrateAndDecryptStorageDocument({
+    document,
+    typeMap: authorizationResult?.typeMap,
+  });
```

Comment:
`get`, first API to use the new `migrateAndDecryptStorageDocument` helper (and also the easiest one to adapt)

Reply:
Now that we're bundling migrate and decrypt together, can we still identify which of these fails if something goes wrong?